x/ethbridge/client/rest/rest.go

```diff
@@ -51,7 +54,8 @@ type burnEthReq struct {
 func RegisterRESTRoutes(cliCtx context.CLIContext, r *mux.Router, storeName string) {
 	r.HandleFunc(fmt.Sprintf("/%s/prophecies", storeName), createClaimHandler(cliCtx)).Methods("POST")
 	r.HandleFunc(fmt.Sprintf("/%s/prophecies/{%s}/{%s}/{%s}/{%s}/{%s}/{%s}", storeName, restEthereumChainID, restBridgeContract, restNonce, restSymbol, restTokenContract, restEthereumSender), getProphecyHandler(cliCtx, storeName)).Methods("GET")
```

Comment:
cc: @denalimarsh

Reply:
This probably makes sense, though do think this should be thought through more - there are a bunch of implications of adding ids, like:
 - more space used
 - doesn't improve UX for end users at all really, only devEX for developers building ontop of peggy, which is a minor efficiency difference anyway. End users will likely eventually interact with these systems through some frontend/UI/dapp, where the chainid/contract/nonce/etc will likely be either hardcoded, or selectable in a dropdown, or remembered from previous usage.
 - having an id and endpoint based on id means that prophecies can no longer be retrieved easily based on deterministically working out the url based on what original request they got triggered by, and so there are likely some consumers based that need this characteristic. we can of course still leave this compound-key endpoint in, in addition to adding a single-key id endpoint, so that those needs are still met
 - clients/fat web apps that are built in future consuming this UI now also may need to keep track of ids, and the id will be generated server(blockchain) side and so they need to get it from the server
 - most transactions/requests will likely come in asynchronously, where it is not possible to return the id in that request, so consumers will need to do a second query later on, or wait for transaction confirmation until they get the id back and know what it is.

anyway, as i said, we can have both query endpoints in so may make sense to add another ID based endpoint and IDs on prophecies as you suggest

Reply:
@musnit those are great questions. Would you mind opening an issue for this with your response? I don't want to block this PR on it. We can discuss it later there